src/features/tokens/CosmNativeTokenAdapter.ts

```diff
@@ -0,0 +1,135 @@
+import { MsgTransferEncodeObject } from '@cosmjs/stargate';
+import Long from 'long';
+
+import {
+  BaseAppAdapter,
+  ChainName,
+  CosmJsProvider,
+  CwHypCollateralAdapter,
+  ITokenAdapter,
+  MinimalTokenMetadata,
+  MultiProtocolProvider,
+  TransferParams,
+} from '@hyperlane-xyz/sdk';
+import { ProtocolType } from '@hyperlane-xyz/utils';
+
+const COSMOS_IBC_TRANSFER_TIMEOUT = 60_000; // 1 minute
+
+// TODO cosmos move everything in this file to the SDK
+export class BaseCosmosAdapter extends BaseAppAdapter {
+  public readonly protocol: ProtocolType = ProtocolType.Cosmos;
+
+  public getProvider(): CosmJsProvider['provider'] {
+    return this.multiProvider.getCosmJsProvider(this.chainName);
+  }
+}
+
+export class CosmNativeTokenAdapter extends BaseCosmosAdapter implements ITokenAdapter {
+  constructor(
+    public readonly chainName: ChainName,
+    public readonly multiProvider: MultiProtocolProvider,
+    public readonly addresses: Record<string, Address>,
+    public readonly properties: {
```

Comment:
Actually this was intentionally generic in case we end up needing a general property bag in the 4th param spot for other adapters, that way we can keep the args more similar to each other